North-Seattle-College/ad440-winter2021-thursday-repo
# Complete automation of ARM template and removing user interactivity (added Login module)

Closes #96

Changes the script to an ARM template json and automates the deployment of the template.

### How to test

1. Pull my fork and branch in order to test the ARM template and script
`git remote set-url origin <url`
`git pull`
`git checkout <branch>`

2. Using the powershell terminal in VSCode you can call my script by using: 
`./script.ps1 -TenantID {} -SPApplicationID {} -SPSecret {} -SubscriptionID {} -resourceGroupName {} -location {} -templateUri ./Automation/FunctionAppUsersUserID/template.json`

Make sure to fill in the {} with the information from the secrets in nsc-kv-usw2-commonpsswds and to follow the correct naming conventions for the resourceGroupName

3. Wait for the script to finish and create everything successfully
4. Go to Azure and check if the resource group was created with the name you provided and there should be an App Service plan, Function App, Application Insights, and Storage Account that was created inside the resource group. 

After confirming everything was successfully created, delete the resource group!
